### ISSUE TITLE
Scale a screenshot (feature request)

### ISSUE THREAD

**Issue by Raytle** [flameshot-org/flameshot#1700]
### I would like an option to scale a screenshot

Currently when I take a screenshot, if the image is too large I need to scale down the image. For example, 
I might need to scale an image down to 60%. Of course it would be handy if Flameshot had an option that enabled me to scale a screenshot, for example, down to 20% or up to 170%.

Of course this would require adding a new button the the Flameshot GUI which I will call `Scale value`. I imagine that the user would **left** mouse click to change this value. Furthermore, I suppose `Scale value` would have a tooltip that might indicate, for example, "This screenshot will be scaled down to 45%" or "This screenshot will be scaled up to 130%".

### I would like an option to set the default value to scale my screenshots

I generally find most screenshots I take with Flameshot are larger than I would like. Therefore, I would like to be able to set the default percentage to scale all of my screenshots. In my case, I would probably set this value such to 60% or 70%.

In addition to being able to modify this value in the configuration panel, I would like to be able to simply **right** mouse click on `Scale value` icon in the Flameshot GUI to change the **default value** to scale all of my screenshots.

### I would like an option replace Flameshot's default splash screen message (`showhelp`) to remind me of the default value which my screenshots will be scaled to

When I startup Flameshot in a GUI, I would like to see a splash screen message in 32 point font size such as,"Currently Flameshot will scale my images down to 70%".

I am imagining something such as..

-  being able to change`flameshot config --showhelp true`to 
- `flameshot config --showhelp false --showdefaultscalepercent true --currentdefaultscalepercent 70 --scalesplashfontsize 32 --scalesplashmessage "Currently Flameshot will scale my images down to "`.

1. `showdefaultscalepercent` true would be analogous to `showhelp`.
2. `currentdefaultscalepercent 70` would set the default scale percentage to 70%.
3. `scalesplashfontsize 32` would set the splash screen font size to 32 points.
4. In conjunction with point number 2 in this list, `scalesplashmessage "Currently Flameshot will scale my images down to"` would, set the splash screen message to, "Currently Flameshot will scale my images down to 70%".

### Notes:

1. To be sure you understand what I am thinking of: when a user clicks on the  button labeled `Scale value`, a message would pop up like, "Please enter a vale from 1 to 1000 to scale your screenshot." For example, a value of **20** would scale a screenshot down to 20% whereas a value of **150** would scale a screenshot up to 150%.
2. Obviously `Scale value` would be set initially to 100% but the user could change the default value from, for example, 1% to 1000 %.
3. To avoid upsetting the devs (developers), I suppose  `flameshot gui -d 2000 -s 70` might act identically to `flameshot gui -d 2000 --scalepercent 70`. That is, I imagine that both would take a screenshot in two seconds and scale down that screenshot to 70%. Personally, I'd like `flameshot gui -delay_taking_a_screenshot_in_milliseconds 2000 --scale_your_screenshot_by_percentage 70` but I realize that would probably **break the space/time continuum.**
4. Following on from my previous facetious remark, this is a quibble. Generally, I tend to prefer verbose terms that are self-explanatory. Therefore, instead of `showhelp` I would prefer something like, `show_help_on_splash_screen` and concomitantly `show_default_scale_percentage_on_splash_screen.` In the case of Flameshot, my rationale is simple yet, I suppose, unlikely to persuade the average dev. Unlike much of the nomenclature in, for example, Python or Javascript code—which a developer will often use over, and over, and over again—nomenclature in Flameshot is likely to be very seldom used. Particularly for seldom used nomenclature, self-explanatory yet verbose naming conventions are more important than terse yet cryptic naming conventions. There. I said it. I realize that probably makes me a heretic. But. Well. That's a risk I am willing to take.

**Comment by mmahmoudian** [comment]
I believe this can be very very easily done by using `convert` command from ImageMagic. Basically piping `flameshot gui -r` into `convert`.

I personally believe this feature is something that is slightly out of the scope of Flameshot as it falls in the post-processing of the image after the screenshot and annotations are done.

**Comment by borgmanJeremy** [comment]
I agree with @mmahmoudian

**Comment by mmahmoudian** [comment]
@Raytle I finally managed to find some time to spend on this.

You can use the `convert` command from ImageMagic to to scale and resize your screenshot. you can either go it on a file as explained on [the ImageMagic website](https://imagemagick.org/script/convert.php) or you can pipe the screenshot from Flameshot to `convert`:

```sh
# to scale based on percentage
flameshot gui -r | convert png:- -resize 20% png:- | display

# to scale to absolute size (will fit to the smallest side)
flameshot gui -r | convert png:- -resize 100x20 png:- | display
flameshot gui -r | convert png:- -resize 100x200000 png:- | display
```

The `display` is just another command of ImageMagic to show the picture. You can remove that and pipe it to another software or you can save to a file:

```sh
# to scale based on percentage
flameshot gui -r | convert png:- -resize 20% png:my_file.png
```

**Comment by Raytle** [comment]
> #### to scale based on percentage
> flameshot gui -r | convert png:- -resize 20% png:my_file.png

Thank you very much! That is what I wanted! I just bound it to a hotkey. It works great!

Look. Ok? Please don't be insulted. You are a "tech guy." You guys are different than "non-tech people." Therefore, you probably don't realize how useful that feature would be to myriad non-tech people. That feature would probably be very, very useful to myriad "non-tech people." Why? Because it is likely that many of them—at least occasionally—want to be able to quickly and easily resize a screenshot, yet they probably don't know how to write a simple Bash script, pipe commands, and have probably never heard of ImageMagick or GraphicsMagick.

As for me, I can create a simple Python script that will enable me to easily change the value from 20%, to for example, 70% or 210%—just before I want to take a screenshot with Flameshot. However, non-tech people—who don't know how to make computer sing, dance, and do somersaults—very probably won't know how to do that.

The following describes a feature which I think would enable non-tech people to take advantage of what I am imagining would be a section in the Flameshot documentation called something like, "Enabling Flameshot to do all sorts of cool and useful stuff." I imagine this section would include a list of commands such as, `flameshot gui -r | convert png:- -resize 20% png:my_file.png` with a brief explanation of what each command does.

Then I imagine a method that will enable users to simply and easily create "user defined icons" in Flameshot. By "user defined icons" I mean an icons similar to the icon associated with the Flameshot marker tool, the icon associated with the Flameshot circle tool, the icon associated with the Flameshot rectangle tool, and so on. The user defined icons I have in mind would be a different color than the default Flameshot icons in order to clearly indicate that they are user defined icons. 

Under `Configuration –> Shortcuts` I envision an option called, "Create a user defined icon". When a user chooses "Create a user defined icon" the user will be able to paste in something like, `flameshot gui -r | convert png:- -resize 20% png:my_file.png`.  In addition, I imagine that the user would also be able to enter verbiage ( wording) into the [tooltip](https://www.google.com/search?q=tooltip&client=ubuntu&hs=bLz&channel=fs&sxsrf=ALeKk02PCCOVhr9lqw7XsWzfh5y0XY2xQg:1626254110356&source=lnms&tbm=isch&sa=X&ved=2ahUKEwiVib_UnOLxAhXPtp4KHcSrBaAQ_AUoAXoECAEQAw&biw=1280&bih=623) which would appear when the user "mouses over" over any user defined icon that user had created. 

Put simply: I urge you to make it easy for non-tech people to take advantage Flameshot's ability to do all sorts of cool and useful stuff particularly when it is used in conjunction with other applications. Sure. Tech guys like you can figure stuff like that out on your own. For you guys it's often an enjoyable challenge. But for non-tech people things are very different. They typically don't enjoy "messing around" with computers. Instead, they see the computer as a powerful yet oftentimes inscrutable tool. As a result they often "make do" with unnecessarily inferior solutions which, in practice, normally means either not taking advantage of an existing feature whatsoever, or spending far too much time to take advantage of an existing feature.

**Comment by mmahmoudian** [comment]
@Raytle I wasn't aware or GraphicsMagick existence. Thanks for mentioning it.

Regarding your points, I agree that we can also have such section on our website. I will be thankful if you can contribute to the Flameshot.org website and add such parts if you have time.

I have already tried to put the code behind a semi-graphical interface in a side-project to help users to use these image manipulations. It is basically a very short an well-documented and commented bash script with a super easy installation process:
https://codeberg.org/mehrad/dmenu_shot#installation

But one thing that worth noting is that Flameshot is now also serving non-linux users (mostly Windows and MacOS), which means many of the bash scripts and tools won't work on they machines (well, MacOS is in better shape). So it should be clear to users that all such cool solutions will not work on non-proper operating systems (i.e Windows).

**Comment by Raytle** [comment]
@Raytle I wasn't aware or GraphicsMagick existence. Thanks for mentioning it.

Based on what I have read, GraphicsMagick is faster than ImageMagick. I don't think I ever used GraphicsMagick; I've hardly ever used ImageMagick.


> Regarding your points, I agree that we can also have such section on our website. I will be thankful if you can contribute to the Flameshot.org website and add such parts if you have time.

You probably think I'm a developer. I'm not. I'm a salesman who barely knows how to code. I have written some very basic Python scripts and Bash scripts but I consider myself to be a non-technical person. I do not like math; I do not like science; I do not like computers. But. Strangely. I am not bad math and science and computers. Essentially I like what technology can do; but technology itself is uninteresting to me. To me a computer is like a shovel or a rake: it's a tool which I use.


> I have already tried to put the code behind a semi-graphical interface in a side-project to help users to use these image manipulations. 

Your side project baffles me. Why not add a button in Flameshot that will open your project for users? You are obviously unfamiliar with how non-technical people like me think. Personally, I loathe suckless' dmenu. I have even toyed around with it a little bit. Generally, I refuse to use it. It is hideous.

> It is basically a very short an well-documented and commented bash script with a super easy installation process:
https://codeberg.org/mehrad/dmenu_shot#installation

I am not surprised that you seem to have not grasped what I am trying to convey to you. After all, you are "tech guy." I have worked with myriad tech guys. You guys love, love, love to focus on building new stuff (that is, adding new features) but you guys typically ignore benefits. Features are not benefits. Mostly you guys simply build stuff you like. If, by happenstance, others like it you are delighted; if not, well you go build some more stuff. But you guys rarely pay much attention to UI/UX, user interface/user interface. Do you know what happens when tech guys build stuff without paying much attention to UI/UX? In my experience, typically non-tech people avoid using such applications.

No. That is not a brilliant insight of mine. Not at all. See, for example, [This Is What Happens When You Let Developers Create UI](https://blog.codinghorror.com/this-is-what-happens-when-you-let-developers-create-ui/)

**"Deep down inside every software developer, there's a budding graphic designer waiting to get out. And if you let that happen, you're in trouble. Or at least your users will be, anyway..."**


> But one thing that worth noting is that Flameshot is now also serving non-linux users (mostly Windows and MacOS), which means many of the bash scripts and tools won't work on they machines (well, MacOS is in better shape). So it should be clear to users that all such cool solutions will not work on non-proper operating systems (i.e Windows).

Based on  [How to Create and Run Bash Shell Scripts on Windows 10](https://www.howtogeek.com/261591/how-to-create-and-run-bash-shell-scripts-on-windows-10/) which seems indicates, "Note that there’s no way to run Windows programs from within the Bash environment. You’re restricted to Linux terminal commands and utilities, just as you would be on a typical Linux system." Therefore, it seems like my suggestion probably won't work on Windows.

Why should Mac and Linux be deprived of a beneficial feature due to a limitation within Microsoft Windows? Is it necessary to make Flameshot identical for Microsoft Windows, Mac, and Linux? Why not allow users add in custom icons to Flameshot. If those customs are useless for Microsoft Windows users then simply explain something like this in the Flameshot documentation: "Because there’s no way to run Microsoft Windows programs from within the Bash environment, Microsoft Windows users are unable to take advantage of this feature."

I have used Microsoft Windows, Mac, and Linux. I dislike them all... very much. But I dislike Linux less than I dislike Microsoft Windows and Mac. I am certainly not interested in punished users of Microsoft Windows. I have nothing against them whatsoever. But I repeat: why should Mac and Linux be deprived of a beneficial feature due to a limitation within Microsoft Windows?

**Comment by mmahmoudian** [comment]
> You probably think I'm a developer.

No I don't, but considering that you are comfortable commenting under issues on Github shows that you know your way around Markdown which is what our website is made of. That is actually the reason I asked you to contribute and in [the other post](https://github.com/flameshot-org/flameshot/issues/1703#issuecomment-879785363) went through the trouble of explaining how to clone the website repo and contribute. It is a very straight-forward process, it is well documented and it will at most take about 5 minutes to setup.

**Comment by Raytle** [comment]
More ignorance.

**Comment by borgmanJeremy** [comment]
Banned.

We have a code of conduct and actively enforce it. You were previously warned.
